src/main/java/spoon/support/reflect/reference/CtReferenceImpl.java

```diff
@@ -24,13 +24,17 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Objects;
+import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import static spoon.reflect.path.CtRole.NAME;
 
 public abstract class CtReferenceImpl extends CtElementImpl implements CtReference, Serializable {
 
 	private static final long serialVersionUID = 1L;
+	private static final Pattern IS_ARRAY_OR_INSTANCE = Pattern.compile("\\[\\]|@");
```

Comment:
I wonder if we could optimize this even further by creating the matcher right away. Otherwise, a new matcher object must be created on each invocation. So, something like this:

```suggestion
	private static final Matcher IS_ARRAY_OR_INSTANCE = Pattern.compile("\\[\\]|@").matches("");
```

And then you call `reset(String)` on the matcher instead of `matches(String)` on the pattern, for example:

```java
IS_ARRAY_OR_INSTANCE.reset(simpleName).replaceAll("");
```

Could you try that and see if we can squeeze out even more performance?

Reply:
Re-using the `Matcher` would make the code thread unsafe. I can look into it a little bit more, but I'd suggest to introduce JMH performance tests then for things like that.

Reply:
> Re-using the Matcher would make the code thread unsafe.

Ah, good point. We could wrap it in a `ThreadLocal`, though. If you have time to try it (in the same fashion you did for this optimization, doesn't have to be a super robust performance test), then go ahead. Otherwise we'll just go with your current optimization.

Reply:
Again, I think the be-all-end-all solution here is to not use regex at all. Using regex for performance-critical code is, in general, not the best solution.

Reply:
On second thought, let's just ignore the matcher reuse optimization for now. We can get back to this later once we have a more robust way to evaluate performance.

Reply:
As @SirYwell said, the only real way to test performance is writing [JMH](https://github.com/openjdk/jmh) tests. But improving performance can't be done with a single PR, but every (small) step helps. So maybe try solutions like avoiding regex in another PR?